src/components/InputWithSearch.tsx

```diff
@@ -0,0 +1,91 @@
+import { View, Text, TextInput, TouchableOpacity } from 'react-native';
+import { styles } from '../styles/inputWithSearch.styles';
+import { useState, useEffect } from 'react';
+import { InputWithSearchProps } from '../types/input.types';
+
+export const InputWithSearch = ({
+  searchValue,
+  placeholder,
+  options,
+  // onSearch,
+  onChange,
+  iconRight,
+  iconClose,
+  iconOpen,
+  label,
+}: InputWithSearchProps) => {
+  const [isShowOptions, setIsShowOptions] = useState<boolean>(true);
+  const [isInputShown, setIsInputShown] = useState<boolean>(false);
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const [data, setData] = useState<{ label: string; value: string }[]>([]);
+
+  useEffect(() => {
+    setData(options);
+  }, []);
+
+  const handleOnChange = (query: string) => {
+    onChange(query);
+    setIsShowOptions(true);
+    const formattedQuery = query.toLowerCase();
+    const filteredData = options.filter((option) => {
+      return contains(option, formattedQuery);
+    });
+
+    setData(filteredData);
+  };
+
+  const handleOnPress = (query: string) => {
+    handleOnChange(query);
+    setIsShowOptions(false);
+  };
+
+  const contains = (option: { label: string; value: string }, query: string) => {
+    const { label } = option;
+    if (label.toLowerCase().includes(query)) {
+      return true;
+    }
+    return false;
+  };
```

Comment:
If this function doesn't use any component code (props, function, etc.) you can move this function outside of the component to avoid function recreation on every rerender